webcrawler.py

```diff
@@ -5,12 +5,21 @@
 import scrapy
 from scrapy.crawler import CrawlerProcess
 from scrapy.http.request import Request
+import re
+
 
 KEYWORDS_FILE = 'textfile.txt'
 SCRAPY_OUTPUT_FILE = 'scrapy-output.json'
 RANKED_OUTPUT_FILE = 'output.csv'
 
 
+def contains_date(url):
+    result= re.findall(r'/(\d{4})/(\d{1,2})/(\d{1,2})/', url)
+    if (len(result) == 0):
+        return False
+    else: return True
```

Comment:
This `if` is unnecessary, as is any `if` where the only thing you do is return `True` or `False`. Directly return the test instead, ie. `return len(result) != 0`.

By Python convention, it turns out that is *also* bad practice, since result is a list and lists are "truthy" - that is, empty lists are considered false. This means you can do `return bool(result)` and be done with it.